src/Cli/dotnet/commands/dotnet-new/NewCommandShim.cs

```diff
@@ -51,17 +51,22 @@ public static int Run(string[] args)
                 RestoreProject = RestoreProject
             };
 
-            return New3Command.Run(CommandName, CreateHost(), logger, callbacks, args, null);
+            var disableSdkTemplates = args.Contains("--debug:disable-sdk-templates");
+
+            return New3Command.Run(CommandName, CreateHost(disableSdkTemplates), logger, callbacks, args, null);
         }
 
-        private static ITemplateEngineHost CreateHost()
+        private static ITemplateEngineHost CreateHost(bool disableSdkTemplates)
         {
-            var builtIns = new AssemblyComponentCatalog(new[]
+            var builtIns = new List<KeyValuePair<Guid, Func<Type>>>(new AssemblyComponentCatalog(new[]
+            {
+                typeof(Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Abstractions.IMacro).Assembly
+            }));
+
+            if (!disableSdkTemplates)
             {
-                typeof(Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Abstractions.IMacro).Assembly,
-                typeof(BuiltInTemplatePackageProviderFactory).Assembly,
-                typeof(FileMetadataSearchSource).Assembly
```

Comment:
`typeof(FileMetadataSearchSource).Assembly` was removed, was it intentional?

We are seeing `dotnet new android` started to fail here: https://github.com/xamarin/xamarin-android/pull/5849

`dotnet new` isn't finding our templates in `dotnet/template-packs`. Would this cause this issue? Thanks!

Reply:
Sorry about breaking you, it was different change that broke you... It will be fixed via https://github.com/dotnet/sdk/pull/16967 just waiting for one more approval from @wli3 since he is more familiar with option workloads. Again sry.

Reply:
No problem, thanks.